src/Windows.Core/PinnedStruct{T}.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) to owners found in https://github.com/AArnott/pinvoke/blob/master/COPYRIGHT.md. All rights reserved.
+// Licensed under the MIT license. See LICENSE.txt file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Runtime.InteropServices;
+
+    /// <summary>
+    /// Allocate a structure on the heap and pin it in place, allowing a controlled release (Via Dispose).
+    /// </summary>
+    /// <typeparam name="T">Type of the structure that will be pinned.</typeparam>
+    public class PinnedStruct<T> : IDisposable
```

Comment:
I didn't know about it.

It would make the code slightly cleaner as it avoid the cast in `Value` but i'll need to experiment on how after pinning the `StrongBox<T>` instance I can get a pointer to the inner `Value` field. Maybe for another PR :)